File: MultiLanguageSandbox/src/test/deps/javascript/25.js

/**
 * Shifts all characters by 5 positions in alphabetical order.
 * Only letters are replaced, and all letters are in uppercase.
 *
 * @param {string} text The input text to be processed.
 * @returns {string} The transformed text with characters shifted by 5 positions.
 *
 * @example
 * shiftCharacters("NS BFW, JAJSYX TK NRUTWYFSHJ FWJ YMJ WJXZQY TK YWNANFQ HFZXJX")
 * // returns "IN WAR, EVENTS OF IMPORTANCE ARE THE RESULT OF TRIVIAL CAUSES"
 */

function shiftCharacters(text) {
    let result = '';
    for (let i = 0; i < text.length; i++) {
        const char = text[i];
        if (char >= 'A' && char <= 'Z') {
            // Calculate the new character after shifting 5 positions
            const originalPos = char.charCodeAt(0) - 'A'.charCodeAt(0);
            const newPos = (originalPos - 5 + 26) % 26; // Subtract 5 and handle negative values by adding 26 before mod
            const newChar = String.fromCharCode(newPos + 'A'.charCodeAt(0));
            result += newChar;
        } else {
            result += char;
        }
    }
    return result;
}
const testShiftCharacters = () => {
    console.assert(shiftCharacters("NS BFW, JAJSYX TK NRUTWYFSHJ FWJ YMJ WJXZQY TK YWNANFQ HFZXJX") === "IN WAR, EVENTS OF IMPORTANCE ARE THE RESULT OF TRIVIAL CAUSES", 'Test 1 failed');
    console.assert(shiftCharacters("N BTZQI WFYMJW GJ KNWXY NS F QNYYQJ NGJWNFS ANQQFLJ YMFS XJHTSI NS WTRJ") === "I WOULD RATHER BE FIRST IN A LITTLE IBERIAN VILLAGE THAN SECOND IN ROME", 'Test 2 failed');
    console.assert(shiftCharacters("IFSLJW PSTBX KZQQ BJQQ YMFY HFJXFW NX RTWJ IFSLJWTZX YMFS MJ") === "DANGER KNOWS FULL WELL THAT CAESAR IS MORE DANGEROUS THAN HE", 'Test 3 failed');
};

testShiftCharacters();

File: MultiLanguageSandbox/src/test/deps/lua/25.lua

--[[
Shifts all characters by 5 positions in alphabetical order. Only letters are replaced, and all letters are in uppercase.

Parameters:
text (str): The input text to be processed.

Returns:
str: The transformed text with characters shifted by 5 positions.

Examples:
shift_characters("NS BFW, JAJSYX TK NRUTWYFSHJ FWJ YMJ WJXZQY TK YWNANFQ HFZXJX")
-- Expected: "IN WAR, EVENTS OF IMPORTANCE ARE THE RESULT OF TRIVIAL CAUSES"
]]--

function shift_characters(text)
    local result = ""
    for i = 1, #text do
        local c = text:sub(i, i)
        if c >= 'A' and c <= 'Z' then
            -- Calculate the new character after shifting 5 positions
            local new_char_code = string.byte(c) - string.byte('A')
            new_char_code = (new_char_code - 5) % 26
            if new_char_code < 0 then
                new_char_code = new_char_code + 26
            end
            local new_char = string.char(new_char_code + string.byte('A'))
            result = result .. new_char
        else
            result = result .. c
        end
    end
    return result
end
function testShiftCharacters()
  local test1 = "NS BFW, JAJSYX TK NRUTWYFSHJ FWJ YMJ WJXZQY TK YWNANFQ HFZXJX"
  local test2 = "N BTZQI WFYMJW GJ KNWXY NS F QNYYQJ NGJWNFS ANQQFLJ YMFS XJHTSI NS WTRJ"
  local test3 = "IFSLJW PSTBX KZQQ BJQQ YMFY HFJXFW NX RTWJ IFSLJWTZX YMFS MJ"
  assert(shift_characters(test1) == "IN WAR, EVENTS OF IMPORTANCE ARE THE RESULT OF TRIVIAL CAUSES")
  assert(shift_characters(test2) == "I WOULD RATHER BE FIRST IN A LITTLE IBERIAN VILLAGE THAN SECOND IN ROME")
  assert(shift_characters(test3) == "DANGER KNOWS FULL WELL THAT CAESAR IS MORE DANGEROUS THAN HE")
end

testShiftCharacters()